File: Data Structures & Algorithms/target-sum/submission-2.cpp
class Solution {
public:
    int findTargetSumWays(vector<int>& nums, int target) {
        return calculate(nums, 0, 0, target);
    }

    int calculate(vector<int>& nums, int index, int current, int target) {
        if (current == target && index == nums.size()) return 1;
        if (index >= nums.size()) return 0;
        int sum = 0;
        sum += calculate(nums, index + 1, current + nums[index], target);
        sum += calculate(nums, index + 1, current - nums[index], target);
        return sum;
    }
};

File: Data Structures & Algorithms/jump-game/submission-2.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        boolean[] jump = new boolean[n];
        jump[0] = true;
        for (int i = 0; i < n; i++) {
            if (!jump[i]) continue;
            int far = nums[i] + i;
            if (far >= n - 1) return true;
            for (int j = i + 1; j <= far; j++) {
                jump[j] = true;
            }
        }
        return jump[n-1];
    }
}
